### PR TITLE
feat(trainer): Adding Softmax option.

### PR DESCRIPTION
**Pull request by Mino1289**
Removing it as an activation function.

### CODE REVIEW

**Comment by Mino1289** [comment]
Softmax should not be implemented with CrossEntropyLoss during training, might be used for inference, but we'll implement it only for this case.
Thus, this PR should be deleted.
